ext/smoke/repos/steps/setup-masters.sh

```diff
@@ -0,0 +1,287 @@
+  on_master ${master_vm} "echo storeconfigs_backend = puppetdb >> ${puppet_conf}"
+  on_master ${master_vm} "echo reports = store,puppetdb >> ${puppet_conf}"
+  echo ""
+  echo ""
+  
+  echo "STEP: Creating route_file with puppetdb terminus for facts"
+  local route_file
+  route_file=`on_master ${master_vm} "puppet master --configprint route_file" | tail -n 1`
+  on_master ${master_vm} "echo --- > ${route_file}"
+  on_master ${master_vm} "echo master: >> ${route_file}"
+  on_master ${master_vm} "echo \"  facts:\" >> ${route_file}"
+  on_master ${master_vm} "echo \"    terminus: puppetdb\" >> ${route_file}"
+  on_master ${master_vm} "echo \"    cache: yaml\" >> ${route_file}"
+  echo ""
+  echo ""
+  
+  echo "STEP: Setting ownership on everything"
+  on_master ${master_vm} 'chown -R puppet:puppet `puppet config print confdir`'
+  echo ""
+  echo ""
+  
+  echo "STEP: Restart puppetserver and perform a puppet run to ensure that facts and reports are sent to PuppetDB"
+  on_master ${master_vm} "puppet resource service puppetserver ensure=stopped"
+  on_master ${master_vm} "puppet resource service puppetserver ensure=running"
+  set +e
+  on_master ${master_vm} "puppet agent -t"
+  local exit_code="$?"
+  set -e
+  if [[ ! "${exit_code}" -eq 2 && ! "${exit_code}" -eq 0 ]]; then
+    echo "Failed to start-up PuppetDB on ${which_master}!"
+    echo "Exiting the script with a failure ..."
+    exit 1
+  fi
+  on_master ${master_vm} "grep 'replace facts' /var/log/puppetlabs/puppetdb/puppetdb.log && echo '### Facts successfully sent to PuppetDB ###'"
+  on_master ${master_vm} "grep 'replace catalog' /var/log/puppetlabs/puppetdb/puppetdb.log && echo '### Reports successfully sent to PuppetDB ###'"
+  echo ""
+  echo ""
+}
+
+function install_puppetdb_from_module() {
+  local master_vm="$1"
+  local which_master=`identify_master ${master_vm}`
+
+  echo "STEP: Install PuppetDB from the module on ${which_master}!"
+  on_master ${master_vm} "puppet module install puppetlabs-puppetdb"
+  local site_pp="/etc/puppetlabs/code/environments/production/manifests/site.pp"
+  FILE="node '\`facter fqdn\`' {
+    class { 'puppetdb::globals':
+    version => '${puppetdb_version}'
+    }
+  include puppetdb
+  include puppetdb::master::config
+  }
+  node default {
+    notify { 'hello': message => 'hello world' }
+  }"
+  on_master ${master_vm} "echo \"$FILE\" > ${site_pp}"
+  # puppet agent -t returns an exit code of 2 if changes are successfully applied
+  set +e
+  on_master ${master_vm} "puppet agent -t"
+  set -e
+  exited="$?"
+  if [[ "$exited" -ne 2 && "$exited" -ne 0 ]]; then
+    echo "Failed to install PuppetDB from the module on ${which_master}!"
+    echo "Exiting the script with a failure ..."
+    exit 1
+  fi
+  echo ""
+  echo ""
+
+  start_puppetdb ${master_vm}
+  echo "Finished installing PuppetDB via. the module on ${which_master}!"
+  echo ""
+  echo ""
+}
+
+function install_puppetdb_from_package() {
```

Comment:
This code could probably be shared with the repos test b/c the set-up from the package is the same (the only part that is different is the installation of the puppetdb packages, which I imagine we could probably pass in as a string).. Dunno if that should be done in this PR but I imagine you can source from another common bash file. I don't care if you do it, just a thought.